### PR TITLE
Enhancement: calendar overlay enhancement

### PR DESCRIPTION
**Pull request by kellygorr**
#### Pull request checklist

- [O] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Enhancing the calendar component.  Created option to have month picker overlayed on top of calendar so that the calendar remains in one column and switches between the two.  Added accessibility in new overlay to mimic OWA calendar.  Fixed issue: When the month picker is the only calendar, and screen is small, calendar would disappear when it should be visible.
Labels: Enhancement

#### Focus areas to test

(optional)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,5 +27,9 @@
     "github": "^7.0.0",
     "gulp": "^3.9.1",
     "yargs": "^6.5.0"
+  },
```
Comment (dzearing) [inline package.json]: these should be removed.

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss [hunk 2]
```diff
@@ -224,7 +224,7 @@ $Calendar-dayPicker-margin: 10px;
 .headerToggleView {
   height: $Calendar-day;
   @include ms-left(0px);
-  position: absolute;
+  position: absolute !important;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: !important should be avoided. What's the reasoning here?

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss [hunk 3]
```diff
@@ -518,6 +625,12 @@ $Calendar-dayPicker-margin: 10px;
 // On smaller screens the month button toggles to the picking months state.
 @media (max-width: 459px) {
 
+  //Stop elements from going off screen
+  //Resize from 440px to 300px
+  .holder{
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: nit; spaces before `{` brackets.

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss [hunk 4]
```diff
@@ -518,6 +625,12 @@ $Calendar-dayPicker-margin: 10px;
 // On smaller screens the month button toggles to the picking months state.
 @media (max-width: 459px) {
 
+  //Stop elements from going off screen
+  //Resize from 440px to 300px
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: this comment is confusing. "Resize" ?

**Comment by dzearing** [comment]
it would be nice to see some screenshots :)

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss (hunk 2 above)
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: Thanks for the feedback!  I found a way around this and will update.

**Comment by kellygorr** [comment]
Attaching screenshots
calendarpt1
calendarpt2 header clicked

File: package.json (hunk 1 above)
Comment (kellygorr) [inline package.json]: Hi!  Just want to clarify
I should remove:
"github": "^7.0.0",
"gulp": "^3.9.1",
"yargs": "^6.5.0"

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss (hunk 4 above)
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: I'll remove that line, it isn't really necessary.

File: package.json (hunk 1 above)
Comment (kellygorr) [inline package.json]: Nevermind!  I didn't notice the additions below the comment.  I will remove, thanks!

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss [hunk 5]
```diff
@@ -486,6 +489,113 @@ $Calendar-dayPicker-margin: 10px;
...
+    // Show the month picker.
+
+    .holder {
+      width: 230px;
+    }
+
+    .monthPicker {
+      display: block;
+    }
+    // Style the the month and year pickers.
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: Typo: the the

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx [hunk 6]
```diff
@@ -119,7 +121,19 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
...
-          <div className={ css('ms-DatePicker-headerToggleView js-showMonthPicker', styles.headerToggleView) } />
+          {
+            showMonthPickerAsOverlay ?
+              <div
+                className={ css('ms-DatePicker-headerToggleView js-showMonthPicker', styles.headerToggleView) }
+                onClick={ () => this._onSelectSwitchCalendar(false) }
+                onKeyDown={ this._onKeyDown.bind(this, () => this._onSelectSwitchCalendar(true)) }
+                aria-label={ strings.monthPickerAriaLabel }
+                role='button'
+                tabIndex={ 0 }
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: Why do we need another clickable header? Should we just have one header and have that become clickable when needed?

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarMonth.tsx [hunk 7]
```diff
@@ -19,6 +19,8 @@ export interface ICalendarMonthProps extends React.Props<CalendarMonth> {
   onNavigateDate: (date: Date, focusOnNavigatedDay: boolean) => void;
   today?: Date;
   highlightCurrentMonth: boolean;
+  showMonthPickerAsOverlay?: boolean;
+  onSelectSwitchCalendar: (focus: boolean) => void;
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarMonth.tsx]: Same comments as for DayPicker, rather just have this component know whether the header is clickable

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarMonth.tsx [hunk 8]
```diff
@@ -75,6 +77,19 @@ export class CalendarMonth extends BaseComponent<ICalendarMonthProps, {}> {
             </span>
           </div>
           <div className={ css('ms-DatePicker-currentYear js-showYearPicker', styles.currentYear) }>{ navigatedDate.getFullYear() }</div>
+          {
+            showMonthPickerAsOverlay ?
+              <div
+                className={ css('ms-DatePicker-headerToggleView js-showYearPicker', styles.headerToggleView) }
+                onClick={ () => this._onSelectSwitchCalendar(false) }
+                onKeyDown={ this._onKeyDown.bind(this, () => this._onSelectSwitchCalendar(true)) }
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarMonth.tsx]: You don't want to bind event handlers in this way, i.e., using anonymous functions inside render. Use @autobind instead, do you mind changing the others? Also in CalendarDay where applicable.

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarPage.tsx [hunk 9]
```diff
@@ -24,6 +24,10 @@ export class CalendarPage extends React.Component<IComponentDemoPageProps, any>
               <CalendarInlineExample isMonthPickerVisible={ false } dateRangeType={ DateRangeType.Day } autoNavigateOnSelection={ false }
                 showGoToToday={ false } />
             </ExampleCard>
+            <ExampleCard title='Inline Calendar with overlayed month picker when header is clicked' code={ CalendarInlineExampleCode }>
+              <CalendarInlineExample isMonthPickerVisible={ true } showMonthPickerAsOverlay={ true } highlightCurrentMonth={ true } dateRangeType={ DateRangeType.Day } autoNavigateOnSelection={ false }
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarPage.tsx]: You don't need isMonthPickerVisible={ true }, right?

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss [hunk 10]
```diff
@@ -531,6 +646,21 @@ $Calendar-dayPicker-margin: 10px;
     .monthPicker {
       display: block;
     }
+
+  }
+
+  // Do not hide Month picker if it is the only calendar
+  .pickerOnlyMonthPickerVisible, .pickerIsMonthPickerOverlayVisible{
```
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: The month picker has a smaller width than the day picker, so when changing between the two it doesn't looks right.  In Christina's comps she changed the width of the month picker to match the date picker.  When I made the overlay version I didn't want to change the css of the other month pickers (the side-by-side calendar version), since the css had always been like that and I didn't want to break anything that was out in the wid, so that is why there were so many duplicate styles.  But... I didn't realize that the surface calendar would have another date picker and it would be the side-by-side calendar version.  Since that is the case I do need to update the other calendars and I don't need to duplicate the styles like I did.  I will consolidate.

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.Props.ts [hunk 11]
```diff
@@ -144,4 +150,14 @@ export interface ICalendarStrings {
...
+
+  /**
+  * Aria-label for the "month picker" button.
+  */
+  monthPickerAriaLabel?: string;
+
+  /**
+  * Aria-label for the "day picker" button.
+  */
+  dayPickerAriaLabel?: string;
```
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.Props.ts]: Yes!  I will add these clarifications.

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx [hunk 12]
```diff
@@ -19,6 +19,9 @@ export interface ICalendarState {
 
   /** The currently selected date in the calendar */
   selectedDate?: Date;
+
+  isMonthPickerVisible?: boolean;
+  isDayPickerVisible?: boolean;
```
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx]: Yes!

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx [hunk 13]
```diff
@@ -47,12 +51,14 @@ export class Calendar extends BaseComponent<ICalendarProps, ICalendarState> impl
...
-
+    super(props);
     let currentDate = props.value && !isNaN(props.value.getTime()) ? props.value : (props.today || new Date());
+
     this.state = {
       selectedDate: currentDate,
-      navigatedDate: currentDate
+      navigatedDate: currentDate,
+      isMonthPickerVisible: this.props.showMonthPickerAsOverlay ? false : this.props.isMonthPickerVisible,
+      isDayPickerVisible: this.props.showMonthPickerAsOverlay ? true : this.props.isDayPickerVisible
```
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx]: Yes, showMonthPickerAsOverlay would take precedence, because both month picker and day picker need to be visible in order for the overlay to work.  I will add some more details about this in the documentation.

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx [hunk 14]
```diff
@@ -99,8 +105,10 @@ export class Calendar extends BaseComponent<ICalendarProps, ICalendarState> impl
           styles.picker,
           styles.pickerIsOpened,
           styles.pickerIsFocused,
-          isMonthPickerVisible && isDayPickerVisible && ('is-monthPickerVisible ' + styles.pickerIsMonthPickerVisible),
-          isMonthPickerVisible && !isDayPickerVisible && ('is-onlymonthPickerVisible ' + styles.pickerOnlyMonthPickerVisible)
+          isMonthPickerVisible && isDayPickerVisible && !showMonthPickerAsOverlay && ('is-monthPickerVisible ' + styles.pickerIsMonthPickerVisible),
+          isMonthPickerVisible && !isDayPickerVisible && !showMonthPickerAsOverlay && ('is-onlymonthPickerVisible ' + styles.pickerOnlyMonthPickerVisible),
+          isMonthPickerVisible && showMonthPickerAsOverlay && ('is-onlymonthPickerVisible ' + styles.pickerIsMonthPickerOverlayVisible),
+          isDayPickerVisible && showMonthPickerAsOverlay && styles.pickerIsDayPickerOverlayVisible
```
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx]: I will look into this.  The main issue was that I was originally keeping the new overlay styles completely separate from the other css, but since that isn't necessary and I will be consolidating, that may fix the issue.

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx [hunk 15]
```diff
@@ -183,6 +195,16 @@ export class Calendar extends BaseComponent<ICalendarProps, ICalendarState> impl
   }
 
   @autobind
+  private _onSelectSwitchCalendar(focus: boolean) {
+    this.setState({
+      isDayPickerVisible: !this.state.isDayPickerVisible,
+      isMonthPickerVisible: !this.state.isMonthPickerVisible
+    });
+
+    focus ? this._focusOnUpdate = focus : null;
```
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx]: Yes, I will update!

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx [hunk 16]
```diff
@@ -49,6 +49,8 @@ export interface ICalendarDayProps extends React.Props<CalendarDay> {
   dateRangeType: DateRangeType;
   autoNavigateOnSelection: boolean;
   today?: Date;
+  showMonthPickerAsOverlay?: boolean;
+  onSelectSwitchCalendar: (focus: boolean) => void;
```
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: Good idea!  I will update.

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx (hunk 6 above)
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: Someone created this div ahead of me to be used to toggle the header. The css had this message:
// Without modifying the Pickadate JS, this transparent
// button is necessary to toggle the month view.
.headerToggleView { ...... }

I'm not sure what that means, but that is why I used it.

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarMonth.tsx (hunk 7 above)
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarMonth.tsx]: Yes!

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarPage.tsx (hunk 9 above)
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarPage.tsx]: Yes. I think it was just left over from a copy and paste of another example.  I will remove.

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarMonth.tsx (hunk 8 above)
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarMonth.tsx]: ok! I will look at @autoBind and fix.

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss [hunk 17]
```diff
@@ -486,6 +489,113 @@ $Calendar-dayPicker-margin: 10px;
...
+    // Size and position of the month and year labels.
+    .month {
+      @include margin-left(12px);
+    }
+
+    .optionGrid {
+      width: 220px;
+      @include margin(10px, 0, 0, 0px);
+    }
+    // This component is only used on small displays.
```
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: I will be consolidating (more details in one of the messages below).  I kept all my new css separate, but it looks like that wasn't necessary.

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss [hunk 18]
```diff
@@ -486,6 +489,113 @@ $Calendar-dayPicker-margin: 10px;
...
+        display: none;
+      }
+      // Show the year picking components.
+      .yearPicker {
+        display: block;
+      }
+    }
+  }
+
+   // Show the day picker over month picker (1 column layout), if enabled
```
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: Yes! Looks like I got the details swapped - I updated the  comment.

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss [hunk 19]
```diff
@@ -486,6 +489,113 @@ $Calendar-dayPicker-margin: 10px;
...
+      padding: 0;
+      display: inline-block;
+    }
+    // Reduce the size of the month buttons.
+    .monthOption,
+    .yearOption {
+      width: 40px;
+      height: 40px;
+      line-height: 38px;
+      font-size: 12px;
```
Comment (kristoferbrown) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: Nitpick: some of these font sizes are already defined in sass variables. Additionally, some of the frequently repeated pixel values could be stored in new sass variables as well.

Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: Thanks!  I will take a look at this and make updates.

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss [hunk 20]
```diff
@@ -531,6 +583,26 @@ $Calendar-dayPicker-margin: 10px;
     .monthPicker {
       display: block;
     }
+
+  }
+
+  // Do not hide Month picker if it is the only calendar
+  .pickerOnlyMonthPickerVisible, .overlayedCalendars{
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: This seems a bit confusing. It seems like in order to have overlayed calendars, it is a requirement that we are showing both the day picker and the month picker, right? So we would never be in a situation where we would be only showing the month picker and also overlayed calendars. Am I missing something?

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx [hunk 21]
```diff
@@ -97,8 +108,10 @@ export class Calendar extends BaseComponent<ICalendarProps, ICalendarState> impl
 
   public render() {
     let rootClass = 'ms-DatePicker';
-    let { firstDayOfWeek, dateRangeType, strings, isMonthPickerVisible, isDayPickerVisible, autoNavigateOnSelection, showGoToToday, highlightCurrentMonth, navigationIcons } = this.props;
-    let { selectedDate, navigatedDate } = this.state;
+    let { firstDayOfWeek, dateRangeType, strings, showMonthPickerAsOverlay, autoNavigateOnSelection, showGoToToday, highlightCurrentMonth, navigationIcons } = this.props;
+    let { selectedDate, navigatedDate, isMonthPickerVisible, isDayPickerVisible } = this.state;
+    let onHeaderClick = showMonthPickerAsOverlay ? this._onHeaderClick : undefined;
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx]: Are you handling keybpoard navigation too?

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx [hunk 22]
```diff
@@ -97,8 +108,10 @@ export class Calendar extends BaseComponent<ICalendarProps, ICalendarState> impl
...
+    let showSmallMonthPicker = isMonthPickerVisible && isDayPickerVisible || showMonthPickerAsOverlay && isMonthPickerVisible ? true : false;
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx]: You will want to use parenthesis here to ensure the precedence is correct. Also  no need for ? true : false, it is implied. Also the condition is a bit confusing to me, under which conditions do you want to show the smaller month picker? Should we standardize the size so we don't need to special-case it? In OWA, the day and month picker are both the same size.

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx [hunk 23]
```diff
@@ -114,15 +115,27 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
...
               <Icon iconName={ getRTL() ? leftNavigationIcon : rightNavigationIcon } />
             </span >
           </div >
-          <div className={ css('ms-DatePicker-headerToggleView js-showMonthPicker', styles.headerToggleView) } />
+          {
+            this.props.onHeaderClick ?
+              <div
+                className={ css('ms-DatePicker-headerToggleView js-showMonthPicker', styles.headerToggleView) }
+                onClick={ () => this._onHeaderClick(false) }
+                onKeyDown={ () => this._onHeaderClick(true) }
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: Why are these passing different focus values? Also, avoid using the lambdas here, create separate functions for these if needed.

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx [hunk 24]
```diff
@@ -232,15 +247,22 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
...
   }
 
+  @autobind
   private _onSelectPrevMonth() {
     this.props.onNavigateDate(addMonths(this.props.navigatedDate, -1), false);
   }
 
   @autobind
+  private _onHeaderClick(focus: boolean) {
+    this.props.onHeaderClick ? this.props.onHeaderClick(focus) : null;
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: let { onHeaderClick } = this.props;
if (onHeaderClick) {
    onHeaderClick(focus);
}

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarMonth.tsx [hunk 25]
```diff
@@ -53,31 +55,42 @@ export class CalendarMonth extends BaseComponent<ICalendarMonthProps, {}> {
...
+                <Icon iconName={ getRTL() ? leftNavigationIcon : rightNavigationIcon } />
+              </span>
+            </div>
+          </div>
+          {
+            this.props.onHeaderClick ?
+              <div
+                className={ css('ms-DatePicker-headerToggleView js-showYearPicker', styles.headerToggleView) }
+                onClick={ () => this._onHeaderClick(false) }
+                onKeyDown={ () => this._onHeaderClick(true) }
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarMonth.tsx]: Same comment as above

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarMonth.tsx [hunk 26]
```diff
@@ -117,24 +130,33 @@ export class CalendarMonth extends BaseComponent<ICalendarMonthProps, {}> {
...
-    if (ev.which === KeyCodes.enter || ev.which === KeyCodes.space) {
-      callback();
-    }
-  }
+  // @autobind
+  // private _onKeyDown(callback: () => void, ev: React.KeyboardEvent<HTMLElement>) {
+  //   if (ev.which === KeyCodes.enter || ev.which === KeyCodes.space) {
+  //     callback();
+  //   }
+  // }
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarMonth.tsx]: Why is this removed?

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarMonth.tsx [hunk 27]
```diff
@@ -117,24 +130,33 @@ export class CalendarMonth extends BaseComponent<ICalendarMonthProps, {}> {
...
+  @autobind
   private _onSelectMonth(newMonth: number) {
     let { navigatedDate, onNavigateDate } = this.props;
     onNavigateDate(setMonth(navigatedDate, newMonth), true);
   }
+
+  @autobind
+  private _onHeaderClick(focus: boolean) {
+    this.props.onHeaderClick ? this.props.onHeaderClick(focus) : null;
+  }
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarMonth.tsx]: Same comment as above regarding destructuring

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss (hunk 20 above)
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.scss]: When @media (max-width: 459px) all the monthpickers are set to display none.  This prevents the month pickers that need to be visible from disappearing.  The "Inline Calendar with month picker and no day picker" in the example is styled with .pickerOnlyMonthPickerVisible, and should not be set to display: none  at @media (max-width: 459px).  The overlayed callendars (styled with .overlayedCalendars) also should not have a month picker that is set to display:none at @media (max-width: 459px).

calendar1

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx (hunk 22 above)
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx]: There were two different types of month pickers in the example.  If a user wants the month picker only (it is the second calendar in the picture below)  it is styled with .pickerOnlyMonthPickerVisible.  It is a large calendar with a grid of 60x60 pixel boxes, and in the css this is the default size.  The other month picker is the smaller one that is beside the daypicker (first calendar in the picture below) it is styled with .pickerIsMonthPickerVisible.  It has a grid of 40x40 pixel boxes that override the default 60 pixel size.  
So, there are two different month picker styles, and I didn't want to adjust the default sizes or anything.

I will use parenthesis and remove the true : false (I added it to be more explicit because all the different month picker styles complicate things)

cal2

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx (hunk 23 above)
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx]: You are right!  And I'll remove arrow functions.

File: packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx (hunk 21 above)
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/Calendar.tsx]: Yes, looks like headerClick is not the right thing to call it.  Changing to _onHeaderSelect to be more in-line with navigation naming conventions.

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarMonth.tsx (hunk 26 above)
Comment (kellygorr) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarMonth.tsx]: Removed when fixing bind() issue, adding back.

**Comment by kellygorr** [comment]
@johannao76 Hi!  I made updates and left notes.  I added the missing calendar functionality (After selecting a month from the month picker).  Also, I will create a task for the other Month picker issue (where the month picker isn't showing the currently selected month).

File: packages/office-ui-fabric-react/src/components/Calendar/CalendarMonth.tsx [hunk 28]
```diff
@@ -53,31 +55,42 @@ export class CalendarMonth extends BaseComponent<ICalendarMonthProps, {}> {
...
+                onKeyDown={ (e) => this._onKeyDown(this._onSelectPrevYear, e) }
+                aria-label={ strings.prevYearAriaLabel }
+                role='button'
+                tabIndex={ 0 }>
+                <Icon iconName={ getRTL() ? rightNavigationIcon : leftNavigationIcon } />
+              </span>
+              <span
+                className={ css('ms-DatePicker-nextYear js-nextYear', styles.nextYear) }
+                onClick={ this._onSelectNextYear }
+                onKeyDown={ (e) => this._onKeyDown(this._onSelectNextYear, e) }
```
Comment (johannao76) [inline packages/office-ui-fabric-react/src/components/Calendar/CalendarMonth.tsx]: Remove lambdas
